qiskit/compiler/assembler.py

```diff
@@ -189,19 +224,30 @@ def _parse_common_args(backend, qobj_id, qobj_header, shots,
             and determines the runtime environment.
 
     Raises:
-        QiskitError: if the memory arg is True and the backend does not support
-            memory. Also if shots exceeds max_shots for the configured backend. Also if
-            the type of shots is not int.
+        QiskitError:
+            - If the memory arg is True and the backend does not support memory.
+            - If ``shots`` exceeds ``max_shots`` for the configured backend.
+            - If ``shots`` are not int type.
+            - If any of qubit or meas lo's, or associated ranges do not have length equal to
+            ``n_qubits``.
+            - If qubit or meas lo's do not fit into perscribed ranges.
     """
     # grab relevant info from backend if it exists
     backend_config = None
+    backend_defaults = None
+    n_qubits = None
     if backend:
         backend_config = backend.configuration()
+        n_qubits = backend_config.n_qubits
         # check for memory flag applied to backend that does not support memory
         if memory and not backend_config.memory:
             raise QiskitError("memory not supported by backend {}"
                               .format(backend_config.backend_name))
 
+        # set defaults for pulse
+        if backend_config.open_pulse:
+            backend_defaults = backend.defaults()
```

Comment:
Better to wrap this in a None check than rely on defaults always being present if open_pulse is on.

Reply:
Addressed in https://github.com/Qiskit/qiskit-terra/pull/6167/commits/79f38dd8ce56c875bdbf5493119224cc9be0fa7c.